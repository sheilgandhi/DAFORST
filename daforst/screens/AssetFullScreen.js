import {
  View,
  Text,
  StyleSheet,
  Image,
  useWindowDimensions,
} from 'react-native';
import React from 'react';
import { COLORS } from '../colors';

const AssetFullScreen = ({ navigation, route }) => {
  const { asset } = route.params;
  const { width } = useWindowDimensions();

  return (
    <View style={styles.container}>
      <Image
        source={{ uri: `data:image/png;base64,${asset.imgUrl}` }}
        style={styles.image}
      />
      <View style={styles.information}>
        <Text style={styles.lines}>
          <Text style={styles.bold}>ARCore ID: </Text>
          {asset.arcore_id}
        </Text>
        <Text style={styles.lines}>
          <Text style={styles.bold}>ARCore Location: </Text>
          {asset.arcore_location}
        </Text>
        <Text style={styles.lines}>
          <Text style={styles.bold}>IPFS ID: </Text>
          {asset.ipfs_cid}
        </Text>
        <Text style={styles.lines}>
          <Text style={styles.bold}>Name: </Text>
          {asset.name}
        </Text>
        <Text style={styles.lines}>
          <Text style={styles.bold}>Owner: </Text>
          {asset.owner}
        </Text>
        <Text style={styles.lines}>
          <Text style={styles.bold}>Description: </Text>
          {asset.description}
        </Text>
      </View>
    </View>
  );
};

export default AssetFullScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  image: {
    width: '100%',
    height: 300,
  },
  information: {
    flex: 1,
    justifyContent: 'center',
    paddingLeft: 20,
    color: COLORS.black,
  },
  line: {
    marginVertical: 15,
  },
  bold: {
    fontWeight: 'bold',
  },
});
