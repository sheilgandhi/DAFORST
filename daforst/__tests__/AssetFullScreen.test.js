import React from 'react';
import renderer from 'react-test-renderer';
import AssetFullScreen from '../screens/AssetFullScreen';

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  setOptions: jest.fn(),
};
const route = {
  params: {
    asset: {
      imgUrl: jest.fn(),
    },
  },
};

describe('AssetFullScreen', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<AssetFullScreen navigation={navigation} route={route} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
