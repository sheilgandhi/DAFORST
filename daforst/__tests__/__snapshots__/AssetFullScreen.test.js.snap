// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetFullScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "data:image/png;base64,function () {
        return fn.apply(this, arguments);
      }",
      }
    }
    style={
      Object {
        "height": 300,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "color": "#000",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 20,
      }
    }
  >
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        ARCore ID: 
      </Text>
    </Text>
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        ARCore Location: 
      </Text>
    </Text>
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        IPFS ID: 
      </Text>
    </Text>
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Name: 
      </Text>
    </Text>
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Owner: 
      </Text>
    </Text>
    <Text>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Description: 
      </Text>
    </Text>
  </View>
</View>
`;
